### PR TITLE
Gh 120 - Containerize ANET

### PR DESCRIPTION
**Pull request by VassilIordanov**
travis script not tested as it is hard to do w/o checking in.

container images work well, docker compose still needs some work, to be finished once travis starts deploying properly

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -60,3 +60,9 @@ addons:
     branches:
       - master
       - candidate_scan
+
+after_success:
+  - if [ "$TRAVIS_BRANCH" == "candidate" ]; then
+    docker login -u "$DOCKER_USERNAME" -p "$DOCKER_PASSWORD";
+    ./gradlew dockerBuildImage dockerPushImage
```
Comment (gjvoosten) [inline .travis.yml]: The two lines inside the `if` should be indented.

File: src/main/java/mil/dds/anet/WaitForDBCommand.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package mil.dds.anet;
+
+import org.skife.jdbi.v2.DBI;
+
+import io.dropwizard.cli.ConfiguredCommand;
+import net.sourceforge.argparse4j.inf.Namespace;
+import mil.dds.anet.config.AnetConfiguration;
+import io.dropwizard.Application;
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/WaitForDBCommand.java]: Remove unused import.

File: src/main/java/mil/dds/anet/WaitForDBCommand.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		.required(false)
+		.setDefault(20)
+		.help("Nb of attempts before giving up. 20 by default");
+
+		subparser.addArgument("-d", "--delay")
+		.dest("dbConnectionDelay")
+		.type(Integer.class)
+		.required(false)
+		.setDefault(500)
+		.help("Delay in ms between attempts. 500 by default");
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/WaitForDBCommand.java]: Indent all the lines starting with `.`.
I'd also make the defaults into `private static final int CONSTANT_X = yy;` and use those here.

File: src/main/java/mil/dds/anet/WaitForDBCommand.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			catch (Exception exception){
+				dbiException = exception;
+			}
+			try{
+				Thread.sleep(delay);
+			}
+			catch(InterruptedException exception){
+				throw new RuntimeException(exception);
+			}
+		} while (dbiException != null && remainingTries-- > 0);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/WaitForDBCommand.java]: The `dbiException != null && ` is unnecessary here.
The code inside the `do … while` could also be rewritten for clarity.

File: src/main/java/mil/dds/anet/WaitForDBCommand.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+			try {
+				jdbi.close(jdbi.open());
+				dbiException = null;
+				break;
+			}
+			catch (Exception exception){
+				dbiException = exception;
+			}
+			try{
+				Thread.sleep(delay);
```
Comment (gjvoosten) [inline src/main/java/mil/dds/anet/WaitForDBCommand.java]: This sleep should be done only when `remainingTries > 0` else you wait unnecessarily after the final failed attempt.

File: src/main/java/mil/dds/anet/WaitForDBCommand.java (hunk 4 above)
Comment (VassilIordanov) [reply src/main/java/mil/dds/anet/WaitForDBCommand.java]: dbiException != null is needed for the happy path : db.open has succeeded w/o error - time to leave the loop and continue.

**Pull request merged**